### DISCUSSION TITLE
Should validation errors on discriminated unions reference the union subtype?

### DISCUSSION THREAD

**Discussion by senko** [pydantic/pydantic#3918] [Question]
If there's a validation error for a sub-model in a tagged union, the error location includes the sub-model name. Is this intentional?

Code:

```
from typing import Union, Literal, List
from typing_extensions import Annotated
from pydantic import BaseModel, validator, Field

class CharField(BaseModel):
    type: Literal["char"]
    txt: str

    @validator('txt')
    def validate_str(cls, v):
        raise ValueError("some error")

class IntField(BaseModel):
    type: Literal["int"]
    value: int = 0

MyField = Annotated[Union[CharField, IntField], Field(discriminator='type')]

class MyModel(BaseModel):
    fields: List[MyField] = []

data = {
  "fields": [
    {
      "type": "char",
      "txt": "whatever"
    }
  ]
}

MyModel(**data)
```

This results in a validation error (that's expected) with a location that references `CharField` (that's unexpected):

```
pydantic.error_wrappers.ValidationError: 1 validation error for MyModel
fields -> 0 -> CharField -> txt
  some error (type=value_error)
```

For comparison, if I use the submodel directly:

```
class MyModel(BaseModel):
    fields: List[CharField] = []
```

the error is like this:
```
pydantic.error_wrappers.ValidationError: 1 validation error for MyModel
fields -> 0 -> txt
  some error (type=value_error)
```

I believe that code that does this is [here](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/fields.py#L1111), it appends the subtype.

Now my question is: is adding the subtype the expected (intentional) behaviour? I can't any specific reference to that in the docs so I'm not sure if that's fine or a small bug. If it's a bug, I'd be happy to make a PR to fix it.

Thanks!

**Comment by sshishov** [comment]
I think that it should be at least configurable
